### ISSUE TITLE
Wavetable VCO - Non Fatal Label Glitch

### ISSUE THREAD

**Issue by therealkitman** [surge-synthesizer/surge-rack#856]
V.2.1.3.0
VCVRAck 2.2.2.

If you start VCV Rack that has a Wavetable VCO with a  "OneShot" wavetable, its label is missing. But it returns if you change tables or zoom in/out.

![image](https://user-images.githubusercontent.com/14888092/220798542-9e1f3ac5-d4f1-47e1-9b54-5b3c9d1a3f1d.png)

**Comment by baconpaul** [comment]
Having a hard time reproducing this. 

Here's my network

<img width="324" alt="Screen Shot 2023-03-04 at 10 11 43 AM" src="https://user-images.githubusercontent.com/13387561/222913870-1ad276ce-ecbb-4949-b730-d034d9b007b8.png">

and when I load, I get a one shot and a correct draw of blip3

am I missing something in the report? Could you share your patch which shows this problem?

Thanks so much!

**Comment by therealkitman** [comment]
Not much to show. I created a blank patch with the VCO as the only plugin. Usually happens if you're zoomed in on the plugin.

For what it's worth I'm using an Nvidia card and as I said it fixes itself after zooming in/out.

Chasing phantoms?

![image](https://user-images.githubusercontent.com/14888092/222935161-0510e7ac-66b2-4406-9673-5433b9e0e405.png)

**Comment by baconpaul** [comment]
well some condition is causing a redraw to not happen. Note that your waveform is also a bit off. I'm not sure what it is. I'll keep hunting while looking at the other tail-of-three-frame issue also.

**Comment by baconpaul** [comment]
Ahh I see a missing invalidation which could cause this. Have a fix for both this and the corrupted tail paint on the pulse one shot which I should be able to get this morning.

If that doesn't do it I am severely into 'no clue' land but it is consistent with what you show even though i don't see it here. (Missing repaints are always cpu and timing specific so can be platform dependent and stuff too)

**Comment by baconpaul** [comment]
OK I pushed a change that I think will fix this - new nightly in about 30 minutes. If you could confirm. 

If it fixes I'll close this; if not i'll bump it to 2.2 because i have no other ideas right now and would like the ship the rest of the fixes :)

**Comment by therealkitman** [comment]
Nightly??? You'll think me an awful duffer but do you mean this one  "SurgeXTRack-2.1.beta.0-nightly-801c035-lin-x64.vcvplugin "?

**Comment by baconpaul** [comment]
Hmmm that version isn’t what I expected to see. Lemme give some more questions tomorrow 

But if the oneshot is fixed that’s great

**Comment by therealkitman** [comment]
Maybe I should not have updated Surge?  I did when I saw the red dot for the Library updates.

![image](https://user-images.githubusercontent.com/14888092/223011025-1ead5499-630f-4c71-b986-0c34e60b44ca.png)

OK, I reinstalled the nightly, and re-ran testing. The "OneShot" label is still robust as well as the Pulse wavetable looking cleaner.

https://user-images.githubusercontent.com/14888092/223011492-04efcd0d-307b-4d3f-8605-4ef681f1c8a5.mp4
